### PR TITLE
Switch to regular role

### PR DESCRIPTION
**Pull request by k0da**
Controller runs under regular role now, it can't excalate priveleges
during SA creation for execution.

Signed-off-by: Dinar Valeev <dinar.valeev@absa.africa>

### CODE REVIEW

File: pkg/terraform/state/deploy.go [hunk 1]
```diff
@@ -70,8 +70,14 @@ func (h *handler) deployCreate(state *v1.State, input *Input) (*v1.Execution, er
 		return exec, err
 	}
 
-	logrus.Debugf("Create - Creating clusterRoleBinding for %s", state.Name)
-	rb, err := h.createClusterRoleBinding([]metaV1.OwnerReference{}, exec.Name, "cluster-admin", sa.Name, namespace)
+	logrus.Debugf("Create - Creating Role for %s", state.Name)
+	role, err := h.createRole([]metaV1.OwnerReference{}, exec.Name, namespace)
+	if err != nil {
+		logrus.Errorf("error creating crb for %s top level %v", state.Name, err)
```
Comment (somaritane) [inline pkg/terraform/state/deploy.go]:
```suggestion
		logrus.Errorf("error creating role for %s top level %v", state.Name, err)
```

File: pkg/terraform/state/deploy.go [hunk 2]
```diff
@@ -70,8 +70,14 @@ func (h *handler) deployCreate(state *v1.State, input *Input) (*v1.Execution, er
...
+		return exec, err
+	}
+	logrus.Debugf("Create - Creating RoleBinding for %s", state.Name)
+	rb, err := h.createRoleBinding([]metaV1.OwnerReference{}, exec.Name, role.Name, sa.Name, namespace)
 	if err != nil {
 		logrus.Errorf("error creating crb for %s top level %v", state.Name, err)
```
Comment (somaritane) [inline pkg/terraform/state/deploy.go]:
```suggestion
		logrus.Errorf("error creating rolebinding for %s top level %v", state.Name, err)
```

File: pkg/terraform/state/deploy.go [hunk 3]
```diff
@@ -274,38 +286,66 @@ func (h *handler) createServiceAccount(name, namespace string) (*coreV1.ServiceA
 	return sa, nil
 }
 
-func (h *handler) createClusterRoleBinding(or []metaV1.OwnerReference, name, role, sa, namespace string) (*rbacV1.ClusterRoleBinding, error) {
+func (h *handler) createRoleBinding(or []metaV1.OwnerReference, name, role, sa, namespace string) (*rbacV1.RoleBinding, error) {
 	meta := metaV1.ObjectMeta{
 		Name:            "crb-" + name,
```
Comment (somaritane) [inline pkg/terraform/state/deploy.go]:
```suggestion
		Name:            "rb-" + name,
```

**Review by somaritane — CHANGES_REQUESTED** [review]

**Comment by k0da** [comment]
@somaritane e2e passing. It also covers state delete (terraform destroy) case

**Comment by somaritane** [comment]
> @somaritane e2e passing. It also covers state delete (terraform destroy) case

@k0da , unfortunately, state deletion doesn't work on my e2e environment, no destroy execution is started when the state is deleted.

**Comment by k0da** [comment]
@somaritane this test passes https://github.com/AbsaOSS/terraform-controller/blob/master/e2e/e2e_test.go#L190

**Comment by somaritane** [comment]
> @somaritane this test passes https://github.com/AbsaOSS/terraform-controller/blob/master/e2e/e2e_test.go#L190

@k0da , my bad, I had the `destroyOnDelete: false` flag in the State object, after setting it to `true` deletion works just fine 👍

**Review by somaritane — APPROVED** [review]
LGTM 👍

**Pull request merged**